pkg/sypgp/sypgp_test.go

```diff
@@ -528,6 +530,313 @@ func TestPrintEntities(t *testing.T) {
+	}
+
+	// Test with the third line
+	output, err = AskQuestion("Question test 3: ")
+	if err != nil {
+		t.Fatal("failed to get response:", err)
+	}
+	fmt.Println(output)
+
+	// We analyze the result
+	if output != "" {
+		t.Fatal("AskQuestion() returned", output, "instead of an empty string")
+	}
+
+	// Test with the final line
+	output, err = AskQuestion("Question test 4: ")
+	if err != nil {
+		t.Fatal("failed to get response:", err)
+	}
+	fmt.Println(output)
+
+	finalAnswer := restAnswer[len(secondAnswer)+2:] // We have to account for 2 "\n"
+	if output != finalAnswer {
+		t.Fatal("AskQuestion() returned", output, "instead of", finalAnswer)
+	}
+}
+
+func TestAskQuestionNoEcho(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	testStr := "test test\ntest2 test2 test2\n\ntest3 test3 test3 test3"
+	testBytes := []byte(testStr)
+
+	// We create a temporary file that will act as stdin
+	testFile, err := ioutil.TempFile("", "inputTest")
+	if err != nil {
+		t.Fatal("failed to create temporary file:", err)
+	}
+	defer testFile.Close()
+	defer os.Remove(testFile.Name())
+
+	// Write the data that AskQuestionNoEcho will later on read
+	_, err = testFile.Write(testBytes)
+	if err != nil {
+		t.Fatal("failed to write to temporary file:", err)
+	}
+
+	// Reposition to the beginning to ensure there is data to read
+	_, err = testFile.Seek(0, 0)
+	if err != nil {
+		t.Fatal("failed to reposition to beginning of file:", err)
+	}
+
+	// Redirect Stdin
+	savedStdin := os.Stdin
+	defer func() {
+		os.Stdin = savedStdin
+	}()
+	os.Stdin = testFile
+
+	// Test AskQuestionNoEcho(), starting with the first line
+	output, err := AskQuestionNoEcho("Test question 1: ")
+	if err != nil {
+		t.Fatal("failed to get output from AskQuestionNoEcho():", err)
+	}
+	fmt.Println(output)
+
+	// Analyze the result
+	firstAnswer := testStr[:strings.Index(testStr, "\n")]
+	restAnswer := testStr[len(firstAnswer)+1:] // Ignore "\n"
+	if output != firstAnswer {
+		t.Fatalf("AskQuestionNoEcho() returned %s instead of %s", output, firstAnswer)
+	}
+
+	// Test with the second line
+	output, err = AskQuestionNoEcho("Test question 2: ")
+	if err != nil {
+		t.Fatal("failed to get output from AskQuestionNoEcho():", err)
+	}
+	fmt.Println(output)
+
+	// We analyze the answer
+	secondAnswer := restAnswer[:strings.Index(restAnswer, "\n")]
+	if output != secondAnswer {
+		t.Fatalf("AskQuestionNoEcho() returned %s instead of %s", output, secondAnswer)
+	}
+
+	// Test with third line
+	output, err = AskQuestionNoEcho("Test question 3: ")
+	if err != nil {
+		t.Fatal("failed to get output from AskQuestionNoEcho():", err)
+	}
+
+	// We analyze the answer
+	if output != "" {
+		t.Fatalf("AskQuestionNoEcho() returned %s instead of an empty string", output)
+	}
+
+	// Test with the final line
+	output, err = AskQuestionNoEcho("Test question 4: ")
+	if err != nil {
+		t.Fatal("failed to get output from AskQuestionNoEcho():", err)
```

Comment:
This seems to use a combination of `t.Fatal`, and `t.Fatalf`, would it be better to make it consistent? If not, then maybe add a space here:

```suggestion
		t.Fatal("failed to get output from AskQuestionNoEcho(): ", err)
```

Or just make all the `Fatal`s consistent, like:

```suggestion
		t.Fatal("failed to get output from AskQuestionNoEcho(): %v", err)
```